File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-8.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        count = 0
        def dfs(root):
            nonlocal count
            if not root:
                return None

            l_res = dfs(root.left)
            count += 1
            if count == k:
                return root.val
            r_res = dfs(root.right)
            return l_res or r_res

        return dfs(root)

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-8.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func rightSideView(root *TreeNode) []int {
    res := []int{}
    if root == nil{
        return res
    }

    q := []*TreeNode{root}

    for len(q) > 0{
        rightside := []int{}
        rightside = append(rightside, q[len(q)-1].Val)
        res = append(res, rightside...)
        qLen := len(q)
        for i := 0; i < qLen; i++{
            curr := q[0]
            q = q[1:]
            if curr.Left != nil{
                q = append(q, curr.Left)
            }
            if curr.Right != nil {
                q = append(q, curr.Right)
            }
        }
    }
    return res
}
